### PR TITLE
Fix the staffie slots extending forever

### LINKED ISSUE #6902: latejoin job select doesn't condense staffie slots
### Labels

[BUG]

### Description

On the latejoin job selection screen, staff assistant slots are all individually checked off instead of being condensed with a +x.

### Steps to reproduce

1. Join a round that's been going on for awhile.

### Expected Behavior

Tables don't get broken.

### Additional Information & Screenshots

https://i.imgur.com/axwyFoR.png

### PR DESCRIPTION
**Pull request by TheTyrant25**
[BUG] [UI]
## About the PR 
Makes the staffie slots show "join as staff assistant" in one square, with 4 filled squares with crosses and with "+X" for the remaining slots. You know, the way it used to.

Also general cleanup to some of the ui code in that general vicinity, it was illegible and extended off the right side of the screen an unreasonable amount. That's actually the majority of the pr, the fix itself was one line (one integer swapped out for one variable, go figure). I did the cleanup so that i could tell what the hell was actually happening, before i found the culprit.

## Why's this needed? 
fixes #6902 , the nearly year old bug.

### CODE REVIEW

**Comment by TheTyrant25** [comment]
how do i test the latejoin T.T

i do not have 7 friends lying around

**Comment by TheTyrant25** [comment]
okay its working but for some reason it is not linked to the issue on the main PR page which is a bit odd

**Comment by NightmareChamillian** [comment]
try having the words "fixes" or "closes" and then the bug's number, that oughta work.

**Comment by TheTyrant25** [comment]
that works, thanks!

File: code/mob/new_player.dm [hunk 1]
```diff
@@ -397,16 +397,46 @@ mob/new_player
...
-
+				// can you believe all these slot appendages were in one line before using nested ternaries? awful.
+				if (i <= c)
+					if (i == 1 && c > shown)
+						slots += {"
+						<div
+						class='latejoin-card latejoin-full'
+						style='border-color: [J.linkcolor]; background-color: [J.linkcolor];'
+						title='Slot filled.'
+						>+[c - maxslots]
```
Comment (ZeWaka) [inline code/mob/new_player.dm:419]: and what about when it's infinite? this will display wrong

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/mob/new_player.dm (hunk 1 above)
Comment (TheTyrant25) [reply code/mob/new_player.dm:419]: Well when it's infinite, `limit` equals -1, but `maxslots` is made equal to `shown`, and is still capped at `5` (+ the free slot). Neither `maxslots` nor `c` should ever become infinite, so it works fine. From what i've tested using this code, it doesn't break either.
unknown

The prudent change is in line 400:
`var/shown = clamp(c, (limit == -1 ? 99 : limit), maxslots)` to
`var/shown = clamp(c, (limit == -1 ? maxslots : limit), maxslots)`
the rest of the code changes are just cleaning it up to not extend off the right side. It uses exactly identical logic that the old system had before, just not in 200 character ternaries, but broken up into if else statements instead. If there is some logical error in there, it wasn't from me.

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**